### PR TITLE
v3 vaccine edit

### PR DESCRIPTION
**Pull request by bionara**
# Description

Rejigs the add vaccine screen(s) as per the new design to gather further data.
 Full requirements here: 
https://www.notion.so/joinzoe/017c3de591604a7b89d0b69e3c37b279?v=0133357df3ed42e1bd42c1e0d7afcf45&p=8c3e4d43cdfe459da2d65f5426b1b94d
Design: https://www.figma.com/file/7TEcX7f6rw3Z0WYR4TpX3R/COVID-Symptom-Study-(App)---MASTER?node-id=6743%3A3918&viewport=-2281%2C354%2C0.5

## How to test locally
Run the corresponding BE branch if not merged https://github.com/zoe/covid-app/pull/387 . If merged, point against staging or use master.

Access page: home > profile > not had test.
Observe no vaccines + "Add Vaccine" button
Tap "Add Vaccine" button. US/GB users should have a blue link to another screen with image examples (SE not).
Add a DOSE ONE vaccine. To start, just add a date and save. This will let you see the incomplete vaccine/doses list, plus the prompt edit.
From here, edit vaccine and complete fields as required. Note changing the date in edit mode will prompt alert on save.
Delete vaccine now at bottom of dose in edit mode only.

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

Vaccine with a dose that is missing a date (easiest way to replicate is to create a dose, save it, then manually delete its date in django admin)
image

Dose 1 complete
image

Both doses complete
image

Example vaccine doc screen
image

## Checklist

- [ ] I have updated mockServer
- [ ] I've added analytics as relevent
- [ ] I have run `npm test`
- [ ] I have run `npm run test:i18n`

## Out of scope and potential follow-up
Approach followed the Formik form model in which one parent form is passed down to the props, and reused props check if first dose and select properties based on that. Would be good to make this less rigid.
Setting second dose to 'no' on edit mode doesn't delete it yet. Ticket created.

### CODE REVIEW

File: src/features/vaccines/AboutYourVaccineScreen.tsx [hunk 1]
```diff
@@ -26,51 +30,66 @@ type Props = {
...
-interface AboutYourVaccineData extends VaccineDateData {}
+const registerSchema = Yup.object().shape({}).concat(VaccineDoseQuestion.schema());
+
+interface AboutYourVaccineData extends VaccineDoseData {}
 
 export const AboutYourVaccineScreen: React.FC<Props> = ({ route, navigation }) => {
   const vaccineService = useInjection<IVaccineService>(Services.Vaccine);
   const coordinator = assessmentCoordinator;
-  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [errorMessage] = useState<string>('');
```
Comment (petermason) [inline src/features/vaccines/AboutYourVaccineScreen.tsx:40]: no setter and the default value is an empty string?

File: assets/index.ts [hunk 2]
```diff
@@ -100,6 +100,10 @@ export const dietStudyPlaybackReady = require('./diet-playback-ready.png');
 // share svg
 export { default as ShareIcon } from './icons/Share';
 
+// Vaccination example sheets
+export const vaccinationExampleUK = require('./vaccines/vaccine-demo-uk.png');
```
Comment (petermason) [inline assets/index.ts:104]: can you check the size of these. Small icons render poorly on higher pixel density devices. Convert to SVG?

File: src/features/vaccines/AboutYourVaccineScreen.tsx (hunk 1 above)
Comment (bionara) [reply src/features/vaccines/AboutYourVaccineScreen.tsx:40]: Removed as it's unused - global error message at bottom instead.

File: assets/index.ts (hunk 2 above)
Comment (bionara) [reply assets/index.ts:104]: good shout! assumed they were png in the image, but are  svg and can be exported. Doing now.

Comment (bionara) [reply assets/index.ts]: Updated those to SVGs

File: src/components/Screen.tsx [hunk 3]
```diff
@@ -117,7 +117,11 @@ export default class Screen extends Component<ScreenProps> {
                 flexGrow: 1,
                 justifyContent: 'space-between',
               }}>
-              <View style={styles.pageBlock}>{this.props.children}</View>
+              {this.props.extendEdges ? (
```
Comment (bionara) [inline src/components/Screen.tsx:120]: This prop was available on screen, but not used at all in the codebase. I am using it on the vaccine info screen, to set <Screen> margin to 0, then wrap some content (but not the full width image), whilst keeping everything within the screen for scrolling.

Comment (petermason) [reply src/components/Screen.tsx:120]: makes sense

**Review by petermason — APPROVED** [review]

**Comment by selvach** [comment]
This is a great peace of work @bionara !

**Pull request merged**